File: backend/core/src/modules/document-versions/document-versions.controller.ts
import { Body, Controller, Delete, Get, Param, Post, Query, Req, UseGuards } from '@nestjs/common';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';
import { DocumentVersionsService } from './document-versions.service';

@UseGuards(JwtAuthGuard)
@Controller('document-versions')
export class DocumentVersionsController {
  constructor(private readonly documentVersionsService: DocumentVersionsService) {}

  @Get()
  findAll(@Req() req: any, @Query('documentId') documentId?: string) {
    if (!documentId) return [];
    return this.documentVersionsService.findAllForUser(req.user?.userId, documentId);
  }

  @Get(':id')
  findOne(@Req() req: any, @Param('id') id: string) {
    return this.documentVersionsService.findOneForUser(req.user?.userId, id);
  }

  @Post()
  create(@Req() req: any, @Body() dto: any) {
    return this.documentVersionsService.createForUser(req.user?.userId, dto);
  }

  @Post(':id/restore')
  restore(@Req() req: any, @Param('id') id: string) {
    return this.documentVersionsService.restoreForUser(req.user?.userId, id);
  }

  @Delete(':id')
  async remove(@Req() req: any, @Param('id') id: string) {
    await this.documentVersionsService.removeForUser(req.user?.userId, id);
    return { success: true };
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/DocumentVersionsController.java
package store.pjcloud.cowrite.core.controller;

import java.util.List;
import java.util.Map;
import java.util.UUID;
import org.springframework.web.bind.annotation.*;
import store.pjcloud.cowrite.core.entity.Document;
import store.pjcloud.cowrite.core.entity.DocumentVersion;
import store.pjcloud.cowrite.core.security.SecurityUtils;
import store.pjcloud.cowrite.core.service.DocumentVersionsService;

@RestController
@RequestMapping("/document-versions")
public class DocumentVersionsController {
    private final DocumentVersionsService service;

    public DocumentVersionsController(DocumentVersionsService service) {
        this.service = service;
    }

    @GetMapping
    public List<DocumentVersion> list(@RequestParam("documentId") UUID documentId) {
        UUID userId = SecurityUtils.requireUserId();
        return service.findAllForUser(userId, documentId);
    }

    @GetMapping("/{id}")
    public DocumentVersion get(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        return service.findOneForUser(userId, id);
    }

    @PostMapping
    public DocumentVersion create(@RequestBody DocumentVersion body) {
        UUID userId = SecurityUtils.requireUserId();
        return service.createForUser(userId, body);
    }

    @PostMapping("/{id}/restore")
    public Document restore(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        return service.restoreForUser(userId, id);
    }

    @DeleteMapping("/{id}")
    public Map<String, Object> delete(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        service.removeForUser(userId, id);
        return Map.of("success", true);
    }
}
